### PR TITLE
AirloopHVAC:DedicatedOutdoorAirSystem HX sizes to 0 air flow when autosized

### LINKED ISSUE #9704: Heat Exchanger in AirLoopHVAC:DedicatedOutdoorAirSystem fails with set point manager not found
Issue overview
--------------
User file on UnmetHours shows an issue with HXs in the DOAS air loop where the set point manager set point was not found yet there is a set point manager at the supply air outlet of the HX. The mixed air manager SPM did not have a reference set point so that was added.

    SetpointManager:Scheduled,
      OA Air Temp Manager 20,   !- Name
      Temperature,             !- Control Variable
      OA Cooling Supply Air Temp Sch,  !- Schedule Name
      AirLoopDOASSplitterInlet;     !- Setpoint Node or NodeList Name

This did not correct the problem in V22.2. In the debugger there was a cryptic severe error that was tracked down to sizing of the DOAS HX. The SystemAirFlow autosizing function did not account for sizing the HX. Correcting this issue solved the original problem.

The history is described in the [UnmetHours post](https://unmethours.com/question/71531/how-to-add-heatexchangerairtoairsensibleandlatent-with-supply-air-outlet-temperature-control-to-airloophvacdedicatedoutdoorairsystem/?answer=82417#post-id-82417).

[SmallOffice_CentralDOAS_HEXadded_MIxedAir.idf.txt](https://github.com/NREL/EnergyPlus/files/9777098/SmallOffice_CentralDOAS_HEXadded_MIxedAir.idf.txt)

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (issue description)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes #9704
 - AirloopHVAC:DedicatedOutdoorAirSystem HX was not sizing correctly or acknowleding the HX outlet node set point when Supply Air Outlet Temperature Control is used.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] ~If feature, test running new feature, try creative ways to break it~
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] ~Check any new function arguments for performance impacts~
 - [ ] ~Verify IDF naming conventions and styles, memos and notes and defaults~
 - [ ] ~If new idf included, locally check the err file and other outputs~

### CODE REVIEW

**Comment by rraustad** [comment]
Disregard initial commits. This branch was created off of a working branch. The correction is in commit [062a026](https://github.com/NREL/EnergyPlus/pull/9705/commits/062a0260809a8daae9b6166c57f71bdbc4856352)

File: src/EnergyPlus/Autosizing/SystemAirFlowSizing.cc [hunk 1]
```diff
@@ -751,8 +751,11 @@ Real64 SystemAirFlowSizer::size(EnergyPlusData &state, Real64 _originalValue, bo
                 if (this->dataHRFlowSizingFlag) { // HX sizing
                     if (this->curOASysNum) {
                         // size to supply air duct flow rate
-                        if (this->finalSysSizing(this->curSysNum).DesOutAirVolFlow > 0.0) {
+                        if (this->airLoopSysFlag && this->finalSysSizing(this->curSysNum).DesOutAirVolFlow > 0.0) {
                             this->autoSizedValue = this->finalSysSizing(this->curSysNum).DesOutAirVolFlow;
+                        } else if (this->outsideAirSys(this->curOASysNum).AirLoopDOASNum > -1) {
+                            auto &thisAirloopDOAS = this->airloopDOAS[this->outsideAirSys(this->curOASysNum).AirLoopDOASNum];
+                            this->autoSizedValue = thisAirloopDOAS.SizingMassFlow / state.dataEnvrn->StdRhoAir;
```
Comment (rraustad) [inline src/EnergyPlus/Autosizing/SystemAirFlowSizing.cc:756]: The defect file has 5 AirloopHVAC objects and 1 AirloopHVAC:DedicatedOutdoorAirSystem which means FinalSysSizing is allocated to 5 and when CurSysNum = 6 (meaning the 6th outdoor air system associated with the DOAS) an array bounds error shows in the debugger. This method is used in other autosizing functions.

**Comment by rraustad** [comment]
In develop release the HX sizes to 0 but does run. The array bounds failure shows in debug.

    Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, HEX, Design Size Nominal Supply Air Flow Rate [m3/s], 0.00000

**Comment by rraustad** [comment]
Defect file now shows HX with same air flow as DOAS cooling coil.

    Component Sizing Information, Coil:Cooling:Water, OA COOLING COIL 1, Design Size Design Air Flow Rate [m3/s], 1.62123
    Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, HEX, Design Size Nominal Supply Air Flow Rate [m3/s], 1.62123

**Comment by nrel-bot-2c** [comment]
@rraustad @myoldmopar it has been 28 days since this pull request was last updated.

**Review by mjwitte — APPROVED** [review]
Works as advertised. Updated branch, will let CI run one more time.
Thanks @rraustad

**Pull request merged**